Conflict in dependencies

AGPBI: {"kind":"error","text":"Program type already present: com.google.android.gms.internal.measurement.zzwo","sources":[{}],"tool":"D8"}
:app:transformDexArchiveWithExternalLibsDexMergerForDebug FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/0.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/1.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/2.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/3.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/5.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/6.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/7.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/8.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/9.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/10.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/11.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/12.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/13.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/14.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/15.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/16.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/17.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/18.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/19.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/20.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/21.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/22.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/23.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/24.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/25.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/26.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/27.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/28.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/29.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/30.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/31.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/32.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/33.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/34.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/35.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/36.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/37.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/38.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/39.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/40.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/41.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/42.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/43.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/44.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/45.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/46.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/47.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/49.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/50.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/51.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/52.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/53.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/54.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/55.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/56.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/57.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/58.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/59.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/60.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/61.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/62.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/63.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/64.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/65.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/66.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/67.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/68.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/69.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/70.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/71.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/74.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/75.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/76.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/77.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/78.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/79.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/80.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/81.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/82.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/83.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/84.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/85.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/86.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/87.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/88.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/89.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/90.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/91.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/92.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/93.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/94.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/95.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/96.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/97.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/98.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/99.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/100.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/101.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/102.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/103.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/104.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/105.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/106.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/107.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/108.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/109.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/110.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/111.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/112.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/113.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/114.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/115.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/116.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/117.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/118.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/119.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/120.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/121.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/122.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/123.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/124.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/125.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/126.jar, /home/yaraki/src/friendlychat-android/android/app/build/intermediates/transforms/dexBuilder/debug/127.jar
  Program type already present: com.google.android.gms.internal.measurement.zzwo

--- a/android-start/app/build.gradle
+++ b/android-start/app/build.gradle
@@ -27,37 +27,42 @@ android {
 }
 
 dependencies {
-    testCompile 'junit:junit:4.12'
+    configurations.all {
+        resolutionStrategy.force 'com.android.support:design:27.1.1'
+    }
+
+    testImplementation 'junit:junit:4.12'
 
     implementation 'com.android.support:design:27.1.1'
-    implementation 'com.github.bumptech.glide:glide:4.3.1'
+    implementation 'com.github.bumptech.glide:glide:4.7.1'
     implementation 'de.hdodenhof:circleimageview:1.3.0'
     implementation 'com.android.support:appcompat-v7:27.1.1'
 
     // Google
-    implementation 'com.google.android.gms:play-services-auth:15.0.0'
+    implementation 'com.google.android.gms:play-services-auth:15.0.1'
 
     // Firebase
-    implementation 'com.google.firebase:firebase-analytics:15.0.0'
-    implementation 'com.google.firebase:firebase-database:15.0.0'
-    implementation 'com.google.firebase:firebase-storage:15.0.0'
-    implementation 'com.google.firebase:firebase-auth:15.0.0'
-    implementation 'com.google.firebase:firebase-config:15.0.0'
-    implementation 'com.google.android.gms:play-services-appinvite:15.0.0'
-    implementation 'com.google.firebase:firebase-messaging:15.0.0'
-    implementation 'com.google.android.gms:play-services-ads:15.0.0'
-    implementation 'com.google.firebase:firebase-appindexing:15.0.0'
+    implementation 'com.google.firebase:firebase-analytics:15.0.2'
+    implementation 'com.google.firebase:firebase-database:15.0.1'
+    implementation 'com.google.firebase:firebase-storage:15.0.2'
+    implementation 'com.google.firebase:firebase-auth:15.1.0'
+    implementation 'com.google.firebase:firebase-config:15.0.2'
+    implementation 'com.google.android.gms:play-services-appinvite:15.0.2'
+    implementation 'com.google.firebase:firebase-messaging:15.0.2'
+    implementation 'com.google.android.gms:play-services-ads:15.0.1'
+    implementation 'com.google.firebase:firebase-appindexing:15.0.1'
+    implementation 'com.google.firebase:firebase-analytics:15.0.2'
 
     // Firebase UI
     implementation 'com.firebaseui:firebase-ui-database:3.3.1'
 
     //Crashlytics
-    implementation 'com.crashlytics.sdk.android:crashlytics:2.9.1'
+    implementation 'com.crashlytics.sdk.android:crashlytics:2.9.2'
 
     // Testing dependencies
     androidTestImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
-    androidTestImplementation 'com.android.support.test:runner:1.0.1'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
     androidTestImplementation 'com.android.support:support-annotations:27.1.1'
 }
 

--- a/android-start/build.gradle
+++ b/android-start/build.gradle
@@ -5,11 +5,11 @@ buildscript {
         jcenter()
         mavenLocal()
         maven { url 'https://maven.fabric.io/public'  }
-        maven { url 'https://maven.google.com'  }
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.1'
-        classpath 'com.google.gms:google-services:3.2.1'
+        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.google.gms:google-services:3.3.0'
         classpath 'io.fabric.tools:gradle:1.24.4'
 
         // NOTE: Do not place your application dependencies here; they belong

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -27,39 +27,43 @@ android {
 }
 
 dependencies {
-    testCompile 'junit:junit:4.12'
+    configurations.all {
+        resolutionStrategy.force 'com.android.support:design:27.1.1'
+    }
+
+    testImplementation 'junit:junit:4.12'
 
     implementation 'com.android.support:design:27.1.1'
-    implementation 'com.github.bumptech.glide:glide:4.3.1'
+    implementation 'com.github.bumptech.glide:glide:4.7.1'
     implementation 'de.hdodenhof:circleimageview:1.3.0'
     implementation 'com.android.support:appcompat-v7:27.1.1'
 
     // Google
-    implementation 'com.google.android.gms:play-services-auth:15.0.0'
+    implementation 'com.google.android.gms:play-services-auth:15.0.1'
 
     // Firebase
-    implementation 'com.google.firebase:firebase-analytics:15.0.0'
-    implementation 'com.google.firebase:firebase-database:15.0.0'
-    implementation 'com.google.firebase:firebase-storage:15.0.0'
-    implementation 'com.google.firebase:firebase-auth:15.0.0'
-    implementation 'com.google.firebase:firebase-config:15.0.0'
-    implementation 'com.google.android.gms:play-services-appinvite:15.0.0'
-    implementation 'com.google.firebase:firebase-messaging:15.0.0'
-    implementation 'com.google.android.gms:play-services-ads:15.0.0'
-    implementation 'com.google.firebase:firebase-appindexing:15.0.0'
-    implementation 'com.google.firebase:firebase-analytics:15.0.0'
+    implementation 'com.google.firebase:firebase-analytics:15.0.2'
+    implementation 'com.google.firebase:firebase-database:15.0.1'
+    implementation 'com.google.firebase:firebase-storage:15.0.2'
+    implementation 'com.google.firebase:firebase-auth:15.1.0'
+    implementation 'com.google.firebase:firebase-config:15.0.2'
+    implementation 'com.google.android.gms:play-services-appinvite:15.0.2'
+    implementation 'com.google.firebase:firebase-messaging:15.0.2'
+    implementation 'com.google.android.gms:play-services-ads:15.0.1'
+    implementation 'com.google.firebase:firebase-appindexing:15.0.1'
+    implementation 'com.google.firebase:firebase-analytics:15.0.2'
 
     // Firebase UI
     implementation 'com.firebaseui:firebase-ui-database:3.3.1'
 
     //Crashlytics
-    implementation 'com.crashlytics.sdk.android:crashlytics:2.9.1'
+    implementation 'com.crashlytics.sdk.android:crashlytics:2.9.2'
 
     // Testing dependencies
-    androidTestCompile 'junit:junit:4.12'
-    androidTestCompile 'com.android.support.test.espresso:espresso-core:3.0.1'
-    androidTestCompile 'com.android.support.test:runner:1.0.1'
-    androidTestCompile 'com.android.support:support-annotations:27.1.1'
+    androidTestImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support:support-annotations:27.1.1'
 }
 
 apply plugin: 'com.google.gms.google-services'

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -5,11 +5,11 @@ buildscript {
         jcenter()
         mavenLocal()
         maven { url 'https://maven.fabric.io/public'  }
-        maven { url 'https://maven.google.com'  }
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.1'
-        classpath 'com.google.gms:google-services:3.2.1'
+        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.google.gms:google-services:3.3.0'
         classpath 'io.fabric.tools:gradle:1.24.4'
 
         // NOTE: Do not place your application dependencies here; they belong

